modules/components/analytics/config/org.wso2.carbon.pc.analytics.config/src/main/java/org/wso2/carbon/pc/analytics/config/clients/StreamAdminServiceClient.java

```diff
@@ -0,0 +1,98 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.pc.analytics.config.clients;
+
+/**
+ * Access DAS EventStreamAdminService and creates the event stream.
+ */
+
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.client.Options;
+import org.apache.axis2.client.ServiceClient;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.carbon.pc.analytics.config.stubs.EventStreamAdminServiceStub;
+import org.wso2.carbon.pc.analytics.config.stubs.EventStreamAdminServiceStub.AddEventStreamDefinitionAsString;
+import org.wso2.carbon.pc.analytics.config.stubs.EventStreamAdminServiceStub.AddEventStreamDefinitionAsStringResponse;
+
+import java.rmi.RemoteException;
+
+public class StreamAdminServiceClient {
+    private final String serviceName = "EventStreamAdminService";
+    private EventStreamAdminServiceStub serviceAdminStub;
+    private String endPoint;
+    ServiceClient serviceClient;
+    Options option;
+    private static final Log log = LogFactory.getLog(StreamAdminServiceClient.class);
+    JSONObject streamDefinitionJsonOb;
+
+    public StreamAdminServiceClient(String backEndUrl, String session, String streamName, String streamVersion,
+            String streamId, String streamNickName, String streamDescription, JSONArray processVariablesJObArr)
+            throws AxisFault {
+        this.endPoint = backEndUrl + "/services/" + serviceName;
+        serviceAdminStub = new EventStreamAdminServiceStub(endPoint);
+
+        // Authenticate stub from sessionCooke
+        serviceClient = serviceAdminStub._getServiceClient();
+        option = serviceClient.getOptions();
+        option.setManageSession(true);
+        option.setProperty(org.apache.axis2.transport.http.HTTPConstants.COOKIE_STRING, session);
+
+        streamDefinitionJsonOb = new JSONObject();
+        try {
+            streamDefinitionJsonOb.put("streamId", streamId);
+            streamDefinitionJsonOb.put("name", streamName);
+            streamDefinitionJsonOb.put("version", streamVersion);
+            streamDefinitionJsonOb.put("nickName", streamNickName);
+            streamDefinitionJsonOb.put("description", streamDescription);
+            JSONArray processVariablesJArr = new JSONArray();
+
+            //setting process variables as payloadData into the eventStream definition
+            streamDefinitionJsonOb.put("payloadData", processVariablesJObArr);
+            log.debug(streamDefinitionJsonOb.toString());
```

Comment:
don't just log variables. log with a descriptive message, so that a problem can be detected only by looking at logs.